### PR TITLE
grc: Remove extra logger in grc/core/platform

### PR DESCRIPTION
**Pull request by sdh11**
This is an update for PR #2424. The extra debug output from GRC was
being generated by another logger instance in grc/core/platform that
did not respect the command line arguments.

This removes the extra logger and sets the logger level in main to
whatever is requested from the command line. See grc/main.py:43 for more
detail.

### CODE REVIEW

**Review by drmpeg — APPROVED** [review]
Tests fine here. How do you set the logging level on the command line?

**Comment by sdh11** [comment]
`gnuradio-companion --log=debug` should dump all of the debug output.

File: grc/main.py
Comment (mormj) [inline grc/main.py:50]: Do you think bringing grc logging into the grc preferences file would be feasible.  It seems that GRC has a lot of functionality for configuring the loggers for each module, which is currently hardcoded, and the command line is very limiting.  The prefs file might provide a more flexible place to specify all the granularity.

Comment (sdh11) [reply grc/main.py:50]: That should be possible. Are you thinking that each module could have different logging levels?

Comment (mormj) [reply grc/main.py:50]: I was just going off the comment `Note: All other modules need to use the 'grc.<module>' convention` which seems to imply that was the original design.  Seems like it could be overkill, in which case a single grc logging level would be fine.  But you and @skoslowski should have a better idea what is appropriate grc logging granularity

Comment (sdh11) [reply grc/main.py:50]: Ah ok. That `grc.<module>` convention needs to be enforced in each of the modules because it makes each of them a child logger of the parent instance in main (`grc`). Because of how Python imports work the `logging.getLogger(__name__)` statement already handles this. I think I left that comment just for reference.

Adding individual logging levels per module is probably overkill, but I have been experimenting with additional levels to increase verbosity (DEBUG, DEBUG1, DEBUG2, ...). That might be something interesting.

**Comment by mbr0wn** [comment]
There's some good ideas in the conversation, but I don't see why we can't merge this and continue the discussion elsewhere. If this PR doesn't get any attention, it'll anyway bitrot so I'm going to hit that merge button.

**Pull request merged**
